modules/admin_manual/pages/configuration/server/occ_commands/app_commands/_ldap_integration_commands.adoc

```diff
@@ -5,24 +5,51 @@ Marketplace URL: {oc-marketplace-url}/apps/user_ldap[LDAP Integration]
 [source,console]
 ----
 ldap
- ldap:check-user               Checks whether a user exists on LDAP.
+ ldap:check-user               Checks whether a user exists on LDAP
  ldap:create-empty-config      Creates an empty LDAP configuration
  ldap:delete-config            Deletes an existing LDAP configuration
+ ldap:invalidate-cache         Invalidates the LDAP cache
  ldap:search                   Executes a user or group search
  ldap:set-config               Modifies an LDAP configuration
  ldap:show-config              Shows the LDAP configuration
  ldap:test-config              Tests an LDAP configuration
 ----
 
+== Search for a User
+
 Search for an LDAP user, using this syntax:
 
 [source,console,subs="attributes+"]
 ----
-{occ-command-example-prefix} ldap:search [--group] [--offset="..."] [--limit="..."] search
+{occ-command-example-prefix} ldap:search [options] [--] <search>
 ----
+=== Arguments
+
+[width="100%",cols="20%,70%",]
+|===
+| `search`
+| The search string (can be empty)
+|===
+
+Note that the search sting can be empty to list all users, or groups when the `--group` option is set. In this case it must be declared with `''`. It is by default a prefix search. Start the string with `*` to change to infix (substring) search.
 
-Searches match at the beginning of the attribute value only.
-This example searches for `givenNames` that start with 'rob':
+Search attributes are configurable in the LDAP wizard (menu:Settings[Admin > User Authentication > LDAP > Advanced > User Search Attributes]. Which are, if not otherwise defined: Search attributes -> ownCloud's displayname attribute (by default) -> "cn" (by default).
```

Comment:
"Search attributes are configurable in the LDAP wizard (menu:Settings[Admin > User Authentication > LDAP > Advanced > User Search Attributes]). If the list of attributes is empty, ownCloud will use whatever attribute is set as ownCloud's displayname attribute (menu:Settings[Admin > User Authentication > LDAP > Advanced > User Display Name Field])
For groups, the behaviour is the same, although they have their own settings: Group Search Attributes and User Display Name Field".

I'm not sure if it's clear enough for the groups, but I didn't want to repeat the same stuff again.

Reply:
Note that for users, the default displayname attribute is "displayname" not "cn". For groups, the default displayname attribute is "cn".

Reply:
I think I am getting closer, will file an update

Reply:
updated, pls have a look.
btw more small issues fixed 😄

Reply:
> Search attributes are configurable in the LDAP wizard or via `ldap:set-config` as described below:
>
> .For users, the default displayname attribute is "displayname"
> menu:Settings[Admin > User Authentication > LDAP > Advanced > User Search Attributes]

It seems a paragraph is missing... maybe "If the search attributes are empty, the default displayname attributes set in the configuration will be used, otherwise only the configured search attributes will be used, in particular, the 'displayname' attribute might need to be explicitly added to the list"

This way, both paragraphs seem more connected.

Reply:
> attributes set in the configuration will be used

in which configuration, where are these configurations located, are the gui settings different to the occ settings? not clear to me. Not today, but we should have a call tomorrow to clarify.
We need to anyways revise the ldap documentation where we also need to address this.

Reply:
I have improved the search attribute description according our discussion - pls have a look